test/utils/executor.hpp

```diff
@@ -146,14 +121,21 @@ class CommonTestFixture : public ::testing::Test {
 #endif
     using index_type = int;
 
-    CommonTestFixture() : ref{gko::ReferenceExecutor::create()}
+    CommonTestFixture()
+        :
+#ifdef GKO_COMPILING_CUDA
+          stream(ResourceEnvironment::cuda_device_id),
+#elifdef GKO_COMPILING_HIP
+          stream(ResourceEnvironment::hip_device_id),
+#endif
+          ref{gko::ReferenceExecutor::create()}
     {
-#if defined(GKO_TEST_NONDEFAULT_STREAM) && \
-    (defined(GKO_COMPILING_CUDA) || defined(GKO_COMPILING_HIP))
+#if defined(GKO_COMPILING_CUDA) || defined(GKO_COMPILING_HIP)
         init_executor(ref, exec, stream.get());
 #else
         init_executor(ref, exec);
 #endif
+        guard = exec->get_scoped_device_id_guard();
```

Comment:
we call some of our kernels directly and not through `exec->run`. Thereby, we don't set the device id for the kernel call. I added the guard here, instead of adding it to all instances where we call the kernels directly.